validation/src/CueVetDiagnosticsProvider.ts

```diff
@@ -0,0 +1,123 @@
+import * as vscode from 'vscode';
+import { DiagnosticProvider } from './DiagnosticsProvider';
+import { spawn } from 'child_process';
+import { writeFileSync, rmSync, mkdtempSync } from 'fs';
+import { join } from 'node:path';
+import { tmpdir } from 'node:os';
+import { randomBytes } from 'crypto';
+
+export class CueVetDiagnosticsProvider implements DiagnosticProvider {
+    private collection: vscode.DiagnosticCollection
+
+    private mockContext = `#Context: {
+        appRevision:    string
+        appRevisionNum: int
+        appName:        string
+        name:           string
+        namespace:      string
+        output:         _
```

Comment:
You can also add `outputs`, which is also common in kubevela cases for traits.

Reply:
Addressed here: https://github.com/kubevela-contrib/vscode-plugin/pull/3/commits/175797f820367b2781bdafefc9e98739c9c8ed75

Reply:
The future plan is to make the output/outputs type actually useful, rather than just 'any'.